lighthouse-core/gather/connections/connection.js

```diff
@@ -95,33 +95,37 @@ class Connection {
       const callback = this._callbacks.get(object.id);
       this._callbacks.delete(object.id);
 
-      if (object.error) {
-        return this._handleRawError(object, callback);
-      }
-      log.formatProtocol('method <= browser OK',
+      // handleRawError returns or throws synchronously; wrap to put into promise chain.
+      return callback.resolve(Promise.resolve().then(_ => {
+        if (object.error) {
+          return this.handleRawError(object.error, callback.method);
+        }
+
+        log.formatProtocol('method <= browser OK',
           {method: callback.method, params: object.result}, 'verbose');
-      callback.resolve(object.result);
-      return;
+        return object.result;
+      }));
     }
     log.formatProtocol('<= event',
         {method: object.method, params: object.params}, 'verbose');
     this.emitNotification(object.method, object.params);
   }
 
   /**
-   * @param {{error: {message: string}}} object
-   * @param {{reject: function(*), method: string}} callback
-   * @private
+   * Handles error responses from the protocol, absorbing errors we don't care
+   * about and throwing on the rest.
+   * @param {{message: string}} error
+   * @param {string} method Protocol method that received the error response.
+   * @protected
```

Comment:
no, no underscore for `@protected` (not that closure is looking at this, really only annotating to keep in the style of the rest of the file)